aws-ssm-parameter/src/main/java/com/amazonaws/ssm/parameter/BaseHandlerStd.java

```diff
@@ -0,0 +1,103 @@
+import software.amazon.cloudformation.proxy.ProgressEvent;
+
+public abstract class BaseHandlerStd extends BaseHandler<CallbackContext> {
+    private static final String RETRY_MESSAGE = "Detected retryable error, retrying. Exception message: %s";
+    private static final String RESOURCE_MISSING_CFN_MESSAGE = "Resource of type '%s' with identifier '%s' was not found.";
+
+    @Override
+    public ProgressEvent<ResourceModel, CallbackContext> handleRequest(final AmazonWebServicesClientProxy proxy,
+                                                                       final ResourceHandlerRequest<ResourceModel> request,
+                                                                       final CallbackContext callbackContext,
+                                                                       final Logger logger) {
+        return handleRequest(
+                proxy,
+                request,
+                callbackContext != null ? callbackContext : new CallbackContext(),
+                proxy.newProxy(SSMClientBuilder::getClient),
+                logger);
+    }
+
+    protected abstract ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            AmazonWebServicesClientProxy proxy,
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            ProxyClient<SsmClient> client,
+            Logger logger);
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleError(
+            final String operation,
+            final Exception exception,
+            final ResourceModel model,
+            final CallbackContext callbackContext,
+            final Logger logger) {
+        // All InternalServerExceptions are retryable
+        if (exception instanceof TooManyUpdatesException) {
+            logger.log(String.format(RETRY_MESSAGE, exception.getMessage()));
+            throw new CfnThrottlingException(operation, exception);
+        }
+
+        if (exception instanceof ParameterNotFoundException) {
+            // If operation is for DeleteHandler throwing CfnNotFoundException would not work,
+            // so throw a failed status with an error code for NotFound.
+            // Reference: https://t.corp.amazon.com/P35512867/communication
+            if (operation.equalsIgnoreCase("aws-ssm-parameter::resource-delete")) {
+                return ProgressEvent.<ResourceModel, CallbackContext>builder()
+                        .resourceModel(model)
+                        .status(OperationStatus.FAILED)
+                        .errorCode(HandlerErrorCode.NotFound)
+                        .message(String.format(RESOURCE_MISSING_CFN_MESSAGE, ResourceModel.TYPE_NAME, model.getName()))
+                        .build();
+            } else {
+                throw new CfnNotFoundException(ResourceModel.TYPE_NAME, model.getName());
+            }
+        }
+
+        throw new CfnGeneralServiceException(operation, exception);
+    }
```

Comment:
I am not sure this is used correctly. HandleError is used to filter out exception - the ones u wanna ignore or retry

Reply:
I was mainline referring to [reference1](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-macie/blob/master/aws-macie-findingsfilter/src/main/java/software/amazon/macie/findingsfilter/CreateHandler.java#L34) and [reference2](https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-macie/blob/master/aws-macie-findingsfilter/src/main/java/software/amazon/macie/findingsfilter/BaseMacieFindingFilterHandler.java#L155)

Can you please let me know if I need to change them? If so can you please provide some reference